app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/helperClasses/FriendsViewModel.kt

```diff
@@ -16,57 +16,50 @@ class FriendsViewModel(private val userId: String, private val database: Databas
-        database.getUserData(userId).thenApplyAsync {
-            val friendsList = it.friendList ?: listOf()
-            // Use a mutable list to store the realFriends
-            val realFriends = mutableListOf<Friend>()
-
-            // Counter for tracking when all friends are loaded
-            var friendsLoaded = 0
-
-            // Iterate through each userId in friendsList
-            for (userId in friendsList) {
-                // Get the username CompletableFuture
-                val usernameFuture = database.getUsername(userId)
-
-                // When the CompletableFuture completes, update the list of friends
-                usernameFuture.whenComplete { username, exception ->
-                    if (exception == null) {
-                        database.getUserData(userId).whenComplete { userdata, exception ->
-                            if (exception == null) {
-                                // Add the new Friend object to the realFriends list
-                                realFriends.add(
-                                    Friend(
-                                        userdata.userId!!,
-                                        username,
-                                        userdata.picture?.let { pic -> Utils.decodePhoto(pic) },
-                                        true,
-                                    ),
-                                )
-
-                                // Increment the friendsLoaded counter
-                                friendsLoaded++
-
-                                // Check if all friends have been loaded
-                                if (friendsLoaded == friendsList.size) {
-                                    // Concatenate the testFriends and realFriends lists
-                                    allFriends = realFriends
-
-                                    // Set the initial value of _friendsList to allFriends.
-                                    _friendsList.postValue(allFriends)
-                                }
-                            } else {
-                                // Handle the exception (e.g., log the error, show a message to the user, etc.)
+    /**
+     * Used to upadte the friend list value with the new friend list give in argument
+     * @param newFriendsList new value of the friendsList
+     */
+    fun loadFriends(newFriendsList: List<String>) {
+        // Use a mutable list to store the realFriends
+        val realFriends = mutableListOf<Friend>()
+
+        // Counter for tracking when all friends are loaded
+        var friendsLoaded = 0
+
+        // Iterate through each userId in friendsList
+        for (userId in newFriendsList) {
+            // Get the username CompletableFuture
+            val usernameFuture = database.getUsername(userId)
+
+            // When the CompletableFuture completes, update the list of friends
+            usernameFuture.whenComplete { username, exception ->

```

Comment:
It is more concise to directly call the function without creating a val that is not used :
```kotlin
database.getUsername(userId).whenComplete { username, exception ->
```